rpc/src/tests.rs

```diff
@@ -0,0 +1,481 @@
+use super::*;
+use jsonrpsee::core::Error;
+use mock::*;
+use pallet_dex::rpc::RpcError;
+use std::sync::Arc;
+
+type AssetId = u32;
+type Balance = u128;
+type AssetBalance = u64;
+type RpcResult<T> = Result<T, RpcError>;
+
+const ASSET: AssetId = 1;
+const CURRENCY_AMOUNT: Balance = 100;
+const TOKEN_AMOUNT: AssetBalance = 100;
+const EXCHANGE_NOT_FOUND_MESSAGE: &str = "Exchange not found";
+const NOT_ENOUGH_LIQUIDITY_MESSAGE: &str = "Not enough liquidity";
+const OVERFLOW_MESSAGE: &str = "Overflow";
+const RUNTIME_ERROR_MESSAGE: &str = "Runtime error";
+
+fn assert_exchange_not_found(error: Error) {
+    assert!(matches!(error, Error::Call(e) if matches!(&e, CallError::Custom(e)
+        if e.code() == EXCHANGE_NOT_FOUND && e.message() == EXCHANGE_NOT_FOUND_MESSAGE && e.data().is_none())));
+}
+
+fn assert_not_enough_liquidity(error: Error) {
+    assert!(matches!(error, Error::Call(e) if matches!(&e, CallError::Custom(e)
+        if e.code() == NOT_ENOUGH_LIQUIDITY && e.message() == NOT_ENOUGH_LIQUIDITY_MESSAGE && e.data().is_none())));
+}
+
+fn assert_overflow(error: Error) {
+    assert!(matches!(error, Error::Call(e) if matches!(&e, CallError::Custom(e)
+        if e.code() == OVERFLOW && e.message() == OVERFLOW_MESSAGE && e.data().is_none())));
+}
+
+fn assert_unexpected(error: Error) {
+    assert!(matches!(error, Error::Call(e) if matches!(&e, CallError::Custom(e)
+        if e.code() == RUNTIME_ERROR && e.message() == RUNTIME_ERROR_MESSAGE && e.data().is_some())));
+}
```

Comment:
These 4 functions could be refactored into a single one, taking `code` and `message` as params.

Reply:
Done